### PR TITLE
Sync v2-6-stable with v2-6-test to release 2.6.3rc1

### PR DESCRIPTION
**Pull request by ephraimbuddy**
Time for `2.6.3rc1`!

### CODE REVIEW

**Review by potiuk — COMMENTED** [review]
Reviewed the commits. LGTM.

**Comment by potiuk** [comment]
Looks like we have Python 3.7 incompatoibility somewhere around triggerer. Only 3.7 tests are failing. One of the side effects of still supporting 3.7 in 2.6 branch :(

**Comment by potiuk** [comment]
I think the problem is this one:  https://github.com/apache/airflow/pull/32411/commits/3b51035b5b4a7823da043637e9db3362e6529658

More information: https://medium.com/@jflevesque/asyncio-exceptions-changes-from-python-3-6-to-3-7-to-3-8-cancellederror-timeouterror-f79945ead378

**Comment by ephraimbuddy** [comment]
> Ah ... easy.. @pytest.mark.asyncio in tests is the problem and it should be converted to the old format. I will try to make a fixup

Awesome, Thanks!

**Comment by ephraimbuddy** [comment]
@potiuk , I wonder if we should just drop the commit? I already made a patch on it to remove walrus use and we still have this different error. It doesn't seem like it's a critical fix too. WDYT?

**Comment by potiuk** [comment]
I was just waiting for my 3.7 image to build (deleted it recently :( and run tests with my fixup @ephraimbuddy .  I pushed fixup - it should work

**Review by eladkal — APPROVED** [review]

File: RELEASE_NOTES.rst [hunk 1]
```diff
@@ -21,6 +21,73 @@
...
+
+Bug Fixes
+"""""""""
+- Use linear time regular expressions (#32303)
+- Fix triggerers alive check and add a new conf for triggerer heartbeat rate (#32123)
+- Catch the exception that triggerer initialization failed (#31999)
+- Hide sensitive values from extra in connection edit form (#32309)
+- Sanitize ``DagRun.run_id`` and allow flexibility (#32293)
+- Add triggerer canceled log (#31757)
+- Fix try_number shown in the ``task?task_id=&dag_id=&execution_date=`` (#32361)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- Fix try number shown in the task view (#32361)
```

File: RELEASE_NOTES.rst [hunk 2]
```diff
@@ -21,6 +21,73 @@
...
+- Fix bugged singleton implementation (#32218)
+- Use ``mapIndex`` to display extra links per mapped task. (#32154)
+- Ensure that main triggerer thread exits if the async thread fails (#32092)
+- Use ``re2`` for matching untrusted regex (#32060)
+- Render list items in rendered fields view (#32042)
+- Fix hashing of ``dag_dependencies`` in serialized dag (#32037)
+- Return ``None`` if an XComArg fails to resolve in a multiple_outputs Task (#32027)
+- Check for DAG ID in query param from url as well as kwargs (#32014)
+- Flash an error message instead of failure in ``rendered-templates`` when map index is not found (#32011)
+- Fix ``ExternalTaskSensor`` when there is not task group TIs for the current execution date (#32009)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- Fix ``ExternalTaskSensor`` when there is no task group TIs for the current execution date (#32009)
```

File: RELEASE_NOTES.rst [hunk 3]
```diff
@@ -21,6 +21,73 @@
...
+- Fix number param html type in trigger template (#31980, #31946)
+- Fix masking nested variable fields (#31964)
+- Fix ``operator_extra_links`` property serialization in mapped tasks (#31904)
+- Decode old-style nested Xcom value (#31866)
+- Add a check for trailing slash in webserver base_url (#31833)
+- Fix connection uri parsing when the host includes a scheme (#31465)
+- This fixes an issue were a database session got stuck (#31128)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- Fix database session closing with ``xcom_pull`` and ``inlets`` (#31128)
```

File: RELEASE_NOTES.rst [hunk 4]
```diff
@@ -21,6 +21,73 @@
...
+"""""""""""""
+- Fix ``task.sensor`` annotation in type stub (#31954)
+- Limit ``Pydantic`` to ``< 2.0.0`` until we solve ``2.0.0`` incompatibilities (#32312)
+- Fix ``Pydantic`` 2 pickiness about model definition (#32307)
+
+Doc only changes
+""""""""""""""""
+- Add explanation about tag creation and cleanup (#32406)
+- Minor updates to docs(#32369,#32315,#32310)
+- Clarify Listener API behavior  (#32269)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- Clarify Listener API behavior (#32269)
```

File: RELEASE_NOTES.rst [hunk 5]
```diff
@@ -21,6 +21,73 @@
...
+Misc/Internal
+"""""""""""""
+- Fix ``task.sensor`` annotation in type stub (#31954)
+- Limit ``Pydantic`` to ``< 2.0.0`` until we solve ``2.0.0`` incompatibilities (#32312)
+- Fix ``Pydantic`` 2 pickiness about model definition (#32307)
+
+Doc only changes
+""""""""""""""""
+- Add explanation about tag creation and cleanup (#32406)
+- Minor updates to docs(#32369,#32315,#32310)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- Minor updates to docs (#32369,#32315,#32310)
```

File: RELEASE_NOTES.rst [hunk 6]
```diff
@@ -21,6 +21,73 @@
...
+- Clarify Listener API behavior  (#32269)
+- Add information for users who ask for requirements (#32262)
+- Add links to DAGRun / DAG / Task in templates-ref.rst (#32245)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- Add links to DAGRun / DAG / Task in Templates Reference (#32245)
```

File: RELEASE_NOTES.rst [hunk 7]
```diff
@@ -21,6 +21,73 @@
...
+- Add comment to warn off a potential wrong fix (#32230)
+- Add a note that we'll need to restart triggerer to reflect any trigger change (#32140)
+- Adding missing hyperlink to the tutorial documentation (#32105)
+- Added difference between Deferrable and Non-Deferrable Operators (#31840)
+- Add comments explaining need for special "trigger end" log message (#31812)
+- Increase readability for  prerequisites.rst (#31794)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]: This one can probably be skipped, or included in the "minor updates" catchall above.

File: RELEASE_NOTES.rst [hunk 8]
```diff
@@ -21,6 +21,73 @@
...
+- Documentation update on Plugin updates. (#31781)
+- Fix SemVer link in security documentation (#32320)
+- Update security model of Airflow (#32098)
+- Update references to restructured documentation from airflow core (#32282)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- Update references to restructured documentation from Airflow core (#32282)
```

**Review by jedcunningham — APPROVED** [review]

**Comment by potiuk** [comment]
:tada:

**Pull request merged**